core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -402,7 +402,7 @@ impl<N: Ord> Peers<N> {
 			Some(p) => p,
 		};
 
-		if peer.view.last_commit.as_ref() >= Some(&new_height) {
```

Comment:
Otherwise we get the neighbor packet from the peer and can't process his latest commit afterwards.

Reply:
hm, yes, but then there is a loophole where a peer could send the same commit message over and over again

Reply:
The second time we receive it then `consider_global` should reject it with `RejectPast` since we updated our view the first time we imported the commit by calling `note_commit_finalized`.

Reply:
This is also consistent with the check for invalid view change in the function above (`update_peer_state`).